src/cmd/commit.ts

```diff
@@ -20,6 +19,66 @@ interface Arguments extends HelmArguments, DroneArgs {
   message?: string
 }
 
+export const bootstrapGit = async (inValues?: Record<string, any>): Promise<void> => {
+  let values = inValues
+  if (!values) {
+    if (isCli) values = (await hfValues({ filesOnly: true })) as Record<string, any>
+    else return
+  }
+  if (!values?.cluster?.domainSuffix) return // too early, commit will handle it
+  const d = terminal(`cmd:${cmdName}:bootstrapGit`)
+  const { remote, branch, email, username, password } = getRepo(values)
+  cd(env.ENV_DIR)
+  process.env.GIT_SSL_NO_VERIFY = '1' // we don't care about this as repo endpoint is either ours or user input
+  let hasCommits = false
+  try {
+    d.debug('Checking out remote into /tmp/xx to test if repo exists and use if needed')
+    // check remote exists by cloning with a 10 second timeout (if remote is unreachable it takes 30 secs to timeout)
+    await $`set +e && rm -rf /tmp/xx >/dev/null && set -e && timeout 10 git clone ${remote} /tmp/xx`
+    // it didn't throw, so we know we have an existing remote
+    // do we have commits remotely?
+    try {
+      await $`cd /tmp/xx && git fetch && git checkout ${branch} && git log && cd -`
+      d.info('We have commits, so we must clone first.')
+      hasCommits = true
+    } catch (e) {
+      d.error(e)
+      cd(env.ENV_DIR) // cd back to be nice
+      if (!`${e}`.includes('would be overwritten by checkout')) {
```

Comment:
Perhaps we should throw error if there is different message than `would be overwritten by checkout'`

Reply:
No, those are expected

Reply:
what does it mean? Please add appropriate comment